### PR TITLE
156 learning task find which bridges cross which rivers

### PR DESCRIPTION
**Pull request by Martin20494**
DESCRIPTION OF PR: Checking the function used for creating dataframe including river and bridge id for each location

## Developer Checklist
- [x] Make code change
- [ ] Update tests
  - [ ] Update / create new tests
  - [ ] Ensure these tests have the expected behaviour
  - [ ] Test locally and ensure tests are passing
- [ ] Update documentation
  - [ ] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

## Reviewer Checklist
- [ ] Check new code for code smells
- [ ] Check new tests
  - [ ] Ensure adequate coverage
  - [ ] Check for code smells within tests
- [ ] Check if documentation needs updating
  - [ ] Readme
  - [ ] Docstrings
  - [ ] Comments
  - [ ] Wiki

### CODE REVIEW

File: src/digitaltwin/bridges_over_rivers.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+) -> gpd.GeoDataFrame:
+
+    # Get the area of interest polygon in well known text format for database querying
+    aoi_wkt= area_of_interest["geometry"][0].wkt
+    crs = area_of_interest.crs.to_epsg()
+
+    # Create bridge dataframe
+    query_bridge = f"""
+    SELECT t50_fid, geometry from nz_bridge_centrelines
+    WHERE ST_INTERSECTS(nz_bridge_centrelines.geometry, ST_GeomFromText('{aoi_wkt}', {crs}));
```
Comment (LukeParky) [inline src/digitaltwin/bridges_over_rivers.py:19]:
```suggestion
    select t50_fid, geometry from nz_bridge_centrelines
    WHERE ST_INTERSECTS(nz_bridge_centrelines.geometry, ST_GeomFromText('{aoi_wkt}', {crs}));
```

File: src/digitaltwin/bridges_over_rivers.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    bridge_river_join = bridge_river_join.rename(
+        columns={
+            't50_fid': 'bridge_id',
+            'river_section_id': 'river_id'
+        }
+    )
+
+    # Re-order columns
+    bridge_river_join = bridge_river_join[['river_id', 'bridge_id', 'geometry']]
+
```
Comment (LukeParky) [inline src/digitaltwin/bridges_over_rivers.py:47]: (demonstrating code review) Add a comment here

File: src/digitaltwin/bridges_over_rivers.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    # Get catchment area
+    catchment_area = get_catchment_area(selected_polygon_gdf, to_crs=2193)
+    my_new_function(catchment_area, engine)
+
+if __name__ == "__main__":
+    sample_polygon = gpd.GeoDataFrame.from_file("selected_polygon.geojson")
+    main(
+        selected_polygon_gdf=sample_polygon,
+        log_level=LogLevel.DEBUG
+    )
```
Comment (LukeParky) [inline src/digitaltwin/bridges_over_rivers.py:91]:
```suggestion
    )

```

File: src/digitaltwin/bridges_over_rivers.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    engine = setup_environment.get_database()
+    # Get catchment area
+    catchment_area = get_catchment_area(selected_polygon_gdf, to_crs=2193)
+    my_new_function(catchment_area, engine)
+
+if __name__ == "__main__":
+    sample_polygon = gpd.GeoDataFrame.from_file("selected_polygon.geojson")
+    main(
+        selected_polygon_gdf=sample_polygon,
+        log_level=LogLevel.DEBUG
```
Comment (LukeParky) [inline src/digitaltwin/bridges_over_rivers.py:90]:
```suggestion
        log_level=LogLevel.INFO
```

**Review by LukeParky — CHANGES_REQUESTED** [review]
Looks good, fix my comments and this is a good demonstration of the PR process

File: src/digitaltwin/bridges_over_rivers.py (hunk 1 above)
Comment (Martin20494) [reply src/digitaltwin/bridges_over_rivers.py:19]: No

**Comment by Martin20494** [comment]
Closing since this was a learning task and we achieved the learning goals and it does not need to be merged into master

**Pull request closed without merging**